octobot_trading/producers/simulator/ohlcv_updater_simulator.py

```diff
@@ -81,18 +83,22 @@ async def handle_timestamp(self, timestamp, **kwargs):
                     elif self.require_last_init_candles_pairs_push:
                         # triggered on first iteration to initialize large candles that might be pushed much later
                         # otherwise but are required to complete TA evaluation
-                        await self.push(time_frame,
-                                        pair,
-                                        [self.last_candles_by_pair_by_time_frame[pair][time_frame.value][-1]],
-                                        partial=True)
-            self.last_timestamp_pushed = timestamp
-            self.require_last_init_candles_pairs_push = False
+                        if time_frame.value in self.last_candles_by_pair_by_time_frame[pair]:
+                            await self.push(time_frame,
+                                            pair,
+                                            [self.last_candles_by_pair_by_time_frame[pair][time_frame.value][-1]],
+                                            partial=True)
         except DataBaseNotExists as e:
             self.logger.warning(f"Not enough data : {e}")
             await self.pause()
             await self.stop()
         except IndexError as e:
             self.logger.warning(f"Failed to access ohlcv_data : {e}")
+        except Exception as e:
+            self.logger.exception(e, True, f"Error when updating from timestamp: {e}")
+        finally:
+            self.last_timestamp_pushed = timestamp
```

Comment:
It's not symbol dependent ?

Reply:
It's data file dependent, each producer has its data file

Reply:
Perfect :+1: